app/validators/placeholders/placeholder_validator.py

```diff
@@ -114,24 +114,25 @@ def validate_answer_type_for_transform(
         if not (
             transform_type in ["format_unit", "format_percentage"]
             and argument_name == "value"
-            and argument.get("source") == "answers"
+            and argument.get("value")
         ):
             return None
 
-        answer_id = argument["identifier"]
-        answer_type = self.questionnaire_schema.answers_with_context[answer_id][
-            "answer"
-        ]["type"]
+        if argument["value"].get("source") and argument["value"]["source"] == "answers":
```

Comment:
This can be checked in the statement above. We don't need 2 seperate ifs.

Reply:
Ok, added one statement that checks all, thanks!